libs/barista-components/filter-field/README.md

```diff
@@ -151,9 +154,14 @@ interface.
 
 The filter-field provides a default implementation of a `DataSource`, named
 `DtFilterFieldDefaultDataSource`, that takes a specific form of data
-(`DtFilterFieldDefaultDataSourceType`). If your data has a different form you
-can either transform your data so the default data source can understand it, or
-you create your own custom data source.
+(`DtFilterFieldDefaultDataSourceType`). Theoretically, the filter-field supports
+an infinite nesting of nodes. Only special nodes like range and distinct need to
+be leaf nodes (the last node of a branch).
```

Comment:
Infinite number of nodes might not be the best description. Maybe something like: 
```suggestion
(`DtFilterFieldDefaultDataSourceType`). The filter-field supports
many layers of nesting for autocomplete nodes. You can have as many autocomplete nodes nested as you need. E.g. Having "continent > country > region > city" as 3 layers of autocomplete works perfectly fine. Nodes that can only be leaf nodes are free-text, range and options. 
```
Distinct options can also have any number of child nodes or layers.